src/org/thoughtcrime/securesms/RegistrationActivity.java

```diff
@@ -428,7 +426,14 @@ protected void onPostExecute(Void result) {
     }
 
     private void updateProviderInfo() {
-        provider = getContext(this).getProviderFromEmailWithDns(emailInput.getText().toString());
+        Util.runOnBackground(() -> {
+            DcProvider newProvider = getContext(this).getProviderFromEmailWithDns(emailInput.getText().toString());
+            Util.runOnMain(()->updateProviderInfo(newProvider));
```

Comment:
it is not-so-good-style to have different functions with the same name doing different things (updateProviderInfo(void) reads things from network - and  updateProviderInfo(param) updates some controls)

i suggest to remove the second function and add it directly as an lambda function; this is maybe not java-style, but makes the flow easier to follow. sth. as

```suggestion
            Util.runOnMain(() -> {
              provider = newProvider;
              if (provider!=null) {
                Resources res = getResources();
                providerHint.setText(provider.getBeforeLoginHint());
                switch (provider.getStatus()) {
                  case DcProvider.DC_PROVIDER_STATUS_PREPARATION:
                    providerHint.setTextColor(res.getColor(R.color.provider_prep_fg));
                    providerLink.setTextColor(res.getColor(R.color.provider_prep_fg));
                    providerLayout.setBackgroundColor(res.getColor(R.color.provider_prep_bg));
                    providerLayout.setVisibility(View.VISIBLE);
                    break;

                  case DcProvider.DC_PROVIDER_STATUS_BROKEN:
                    providerHint.setTextColor(res.getColor(R.color.provider_broken_fg));
                    providerLink.setTextColor(res.getColor(R.color.provider_broken_fg));
                    providerLayout.setBackgroundColor(getResources().getColor(R.color.provider_broken_bg));
                    providerLayout.setVisibility(View.VISIBLE);
                    break;

                  default:
                    providerLayout.setVisibility(View.GONE);
                    break;
                }
              } else {
                providerLayout.setVisibility(View.GONE);
              }
            });
```